mesonbuild/dependencies/data/CMakeListsLLVM.txt

```diff
@@ -1,8 +1,24 @@
+# fail noisily if attempt to use this file without setting:
+# cmake_minimum_required(VERSION ${CMAKE_VERSION})
+# project(... LANGUAGES ...)
+
+cmake_policy(SET CMP0000 NEW)
 
 set(PACKAGE_FOUND FALSE)
 
+list(REMOVE_DUPLICATES LLVM_MESON_VERSIONS)
+
 while(TRUE)
-  find_package(LLVM REQUIRED CONFIG QUIET)
+  #Activate CMake version selection
+  foreach(i IN LISTS LLVM_MESON_VERSIONS)
+    find_package(LLVM ${i}
+    CONFIG
+    NAMES ${LLVM_MESON_PACKAGE_NAMES}
+    QUIET)
```

Comment:
This could be indented a bit for better readability

Reply:
> You should be able to fix the errors in the bionic CI by adding `linux-bionic-gcc` to `skip_on_jobname` in the `test.json`.

@mensinda I tried, but then some tests with LLVM static linking shows UNEXPPASS on Bionic.

Reply:
@mensinda indentation done.

Reply:
You are right, the current matrix implementation does not support this specific use-case...

It looks like the best solution would be to add (or even better move) the `skip_on_*` fields section. Ideally the `skip_on_*` would work similar to the [`exclude`](https://github.com/mesonbuild/meson/blob/master/run_project_tests.py#L880-L896) code, where single cells in the matrix can be excluded. See https://mesonbuild.com/Contributing.html#configuring-project-tests

Format proposal for skipping:

```json
{
  "env": {
    "VAR": "VAL"
  },
  "installed": [
    { "type": "exe", "file": "usr/bin/testexe" },
    { "type": "pdb", "file": "usr/bin/testexe" },
    { "type": "shared_lib", "file": "usr/lib/z", "version": "1.2.3" },
  ],
  "matrix": {
    "options": {
      "opt1": [
        { "val": "abc"   },
        { "val": "qwert" },
        { "val": "bad"   }
      ],
      "opt2": [
        { "val": null    },
        { "val": "true"  },
        { "val": "false" },
      ]
    },
    "skip": [
      {
        "options": { "opt1": "qwert", "opt2": "false" },
        "skip_on_jobname": "foo"
      },
      {
        "options": { "opt1": "bad" },
        "skip_on_env": "VAR1",
        "skip_on_os": "unix"
      }
    ]
  },
  "tools": {
    "cmake": ">=3.11"
  }
}
```